go/arrow/math/type_test.go.tmpl

```diff
@@ -33,6 +33,14 @@ func Test{{.Name}}Funcs_Sum(t *testing.T) {
 	assert.Equal(t, res, {{.Type}}(49995000))
 }
 
+func Test{{.Name}}Funcs_SumEmpty(t *testing.T) {
+	b := array.New{{.Name}}Builder(memory.NewGoAllocator())
+	vec := b.New{{.Name}}Array()
+	b.Release()
```

Comment:
I'd put it at line 38 as a `defer b.Release()` instead.
also: `vec` is leaked. we should probably have a `defer vec.Release()` somewhere as well.

perhaps we should use a CheckedAllocator:

```go
mem := memory.NewCheckedAllocator(memory.NewGoAllocator())
defer mem.AssertSize(t, 0)
```